### ISSUE TITLE
[DataTable] TableToolbarSearch placeholder text not customizable / translatable to multiple variations within one app

### ISSUE THREAD

**Issue by LongBu** [IBM/carbon-components-react#1328]
Use this template if you want to request a new feature, or a change to an existing feature.

If you are reporting a bug or problem, please use the bug template instead.

### Summary

Please describe your request in one or two sentences.

DataTable / TableToolbarSearch =>translateWithId doesn't support multiple placeholder text within the search input within the same react app.  Not all search placeholder text are all alike in the same app, but if there is a workaround the translateWithId functionality, I'd be open to hearing about it.  

Clarify if you are asking for both design and development, or just design, or just development.

### Justification: See the previous sentences.  This likely only necessitates development, but I'm betting the documentation will need to be amended.  

Provide the business reasons for this request.

This is will have impact on IBM

### Desired UX and success metrics

Describe the full user experience for this feature. Also define the metrics by which we can measure success/failure for the user.  

See sentences above.  

### "Must have" functionality

Highlight any "must have" needs and functionality for the request.

See sentences above.  

This should not be a full list of functionality; the Carbon team will work with you to define functionality based on the desired UX.

### Specific timeline issues / requests

Do you want this work within a specific time period? Is it related to an upcoming release?

Yes and yes.  If this could be done within the next month that would be ideal. 

_NB: The Carbon team will try to work with your timeline, but it's not guaranteed. The earlier you make a request in advance of a desired delivery date, the better!_

### Available extra resources

What resources do you have to assist this effort?

I can help if necessary.  

_Carbon is a collaborative system. We encourage teams to build components and submit them for integration as either add-ons or core components._

**Comment by joshblack** [comment]
Hi @LongBu! Could you just change the `translateWithId` prop that you pass into the component in different areas of the app? Agreed that this makes no sense in the context of the app, if you have suggestions for how to do the message formats let me know!

**Comment by LongBu** [comment]
Hi @joshblack, 
In changing the translateWithId prop, this would manifest in different areas of the app where DataTable/ TableToolbarSearch were located.  There might be a good reason for translateWithId, but my suggestion would be to have a propsuch as placeHolderText similar to what is found in the Search Component.  That seems to be the most logical fix, but this unfortunately could break others code on update/re-build, though the amending one's build  would likely not take a large degree of man hours.  Thoughts?
Thanks!

**Comment by joshblack** [comment]
@LongBu agreed that `translateWithId` could have a better 1:1 mapping with explicit text. The reason we have `translateWithId` is so that you could supply one function to DataTable and then it could pass it around the entire component tree. In other words, in cases where you didn't have direct access to a label you could use this method to have deep access for translation 👍

**Comment by LongBu** [comment]
@joshblack 
Can you supply an example of utilizing a function that allows one to change up the placeholderText to something besides one value per project?  

Harnessing 'react-i18next' and injecting InjectedI18nProps, InjectedTranslateProps seems to only way that I've been able to get translatedWithId to accept anything (eg: a direct mapping to one value in an external file): 

<DataTable
 translateWithId={t}
  ....
  />
Maybe I'm missing something or is it just something to live with?
Thanks

**Comment by asudoh** [comment]
Weighing in - Do anybody interested in a "hybrid" approach, where `translateWithId` runs _only_ if e.g. `placeHolderText` is `undefined`? Thanks!
